### ISSUE TITLE
File associations not working for AppImage builds

### ISSUE THREAD

**Issue by tom572** [electron-userland/electron-builder#6664]
* **Electron-Builder Version**: 22.14.13

* **Node Version**: v16.14.0

* **Electron Version**: 17.0.1

* **Electron Type (current, beta, nightly)**: current

* **Target**: AppImage (tested on ubuntu 18.04)

When defining [file associations](https://www.electron.build/configuration/configuration#overridable-per-platform-options) for opening custom file extensions within the application it does not correctly associate them for AppImage builds. This looks to be due to packaging the mime-type into the wrong location in the AppImage for installing the AppImage using AppImageLauncher (the recommended install method). It is unclear why this works, but if the mime-type file is at `/usr/share/mime/packages` rather than `/usr/share/mime`, the mime type gets installed correctly. Interestingly, in .deb builds it seems to be added to `/usr/share/mime/packages` already. This is also the location that the mime-type file exists within the Freecad AppImage where file associations work as intended.

[Here](https://github.com/tom572/electron-builder-app-image-mime-reproducer) is a reproducer using just the electron quick start application and electron builder. On builds other than AppImage, files with extension `example` get associated correctly. On AppImage builds they do not. If you extract the AppImage using [appimagetool](https://appimage.github.io/appimagetool/) and move the mime-type file from `/usr/share/mime` to `/usr/share/mime/packages` and repackage the image, you will see the file association work correctly.

It looks as though the issue exists at this [line](https://github.com/develar/app-builder/blob/c79345743b794ccef41493decafb0b5b2eece505/pkg/package-format/appimage/appLauncher.go#L20).

**Comment by mmaietta** [comment]
Oomph, if the issue is upstream with app-builder, then you'll need to open an issue up there. There's not much I can do from electron-builder as I only maintain this package.

**Comment by tom572** [comment]
Yeah, in hindsight I should have raised it there initially. I have raised an issue against app-builder now, thanks.
